src/main/resources/database/2021-10-05/12-edit-passenger.sql

```diff
@@ -0,0 +1,6 @@
+--liquibase formatted sql
+--changeset rmecwaldowski:4
+
+   ALTER TABLE passenger
+         ADD CONSTRAINT passenger_id
+         FOREIGN KEY (id) REFERENCES reservation(id)
```

Comment:
same are you using right id for FOREIGN KEY (id)?
and you need additional column in passanger for FK